server/src/configuration_set.rs

```diff
@@ -68,18 +72,25 @@ impl ConfigurationSet {
         let include_built_in = parse_include_built_in(value);
 
         // Create the SliceConfig and CompilationState
-        let mut slice_config = SliceConfig::default();
-        slice_config.set_workspace_root_path(root_path.to_owned());
-        slice_config.set_built_in_slice_path(include_built_in.then(|| built_in_path.to_owned()));
-        slice_config.set_search_paths(paths);
-
+        let slice_config = SliceConfig {
+            paths,
+            workspace_root_path: Some(root_path.to_owned()),
+            built_in_slice_path: include_built_in.then(|| built_in_path.to_owned()),
+        };
         let compilation_data = CompilationData::default();
-        Self { slice_config, compilation_data }
+        Self { slice_config, compilation_data, cached_slice_options: None }
     }
 
     pub fn trigger_compilation(&mut self) -> Vec<Diagnostic> {
+        // Re-compute the `slice_options` we're going to pass into the compiler, if necessary.
+        if self.cached_slice_options.is_none() {
+            let mut slice_options = SliceOptions::default();
+            slice_options.references = self.slice_config.resolve_paths();
+            self.cached_slice_options = Some(slice_options);
+        }
+        let slice_options = self.cached_slice_options.as_ref().unwrap();
```

Comment:
```suggestion
        let slice_options = self.cached_slice_options.get_or_insert_with(|| {
            let mut slice_options = SliceOptions::default();
            slice_options.references = self.slice_config.resolve_paths();
            slice_options
        });
```

Here is a cool little way to simplify this. There are [other functions for modifying options in place](https://doc.rust-lang.org/std/option/#modifying-an-option-in-place) that are useful too.